draft-nottingham-public-resolver-errors.md

```diff
@@ -35,28 +35,59 @@ author:
 
-This draft defines a mechanism to communicate such details when DNS resolver filtering of a name is legally mandated, based upon the structured error data for DNS responses introduced by {{!I-D.ietf-dnsop-structured-dns-error}}.
-
-Allowing DNS resolvers to inject user-visible messages brings unique challenges. Because DNS resolvers are often automatically configured by unknown networks and DNS responses are unauthenticated, these messages can come from untrusted parties -- including attackers (e.g., the so-called "coffee shop" attack) that leverage many users' lack of a nuanced model of the trust relationships between all of the parties that are involved in the service they are using.
-
-Furthermore, lowering the barrier to the presentation of messages explaining why access has been denied by the DNS resolver risks encouraging the wider deployment of DNS-based censorship on the Internet.
-
-This draft attempts to mitigate these risks by minimising the information carried in the DNS response to abstract, publicly registered identifiers -- the DNS Resolver Operator ID and the Filtering Incident ID. A consuming party (e.g., a Web browser) can selectively present messages from those operators that they believe to be using this mechanism for its stated goal -- in particular, those who using it to surface policy-driven filtering, rather than enact discretionary censorship or attack end users.
+Some organizations, such as Lumen, monitor legally-mandated filtering as a
+public good, and track filtering incidents in a publicly accessible databases.
+Public resolvers themselves may also choose to track filtering requests over
+time and make them available.
+
+This draft defines a mechanism to communicate an identifier in a database of
+filtering incidents when a DNS resolver filtering of a name is legally mandated,
+based upon the structured error data for DNS responses introduced by
+{{!I-D.ietf-dnsop-structured-dns-error}}.
+
+A consuming party (e.g., a Web browser) can use the identifier to construct a
+link to the specific entry in the database provider. This enables user agents to
+direct users to a location with additional context about why the filtering was
+required.
+
+Allowing DNS resolvers to inject links or user-visible messages brings unique challenges.
+Because DNS resolvers are often automatically configured by unknown networks and
+DNS responses are unauthenticated, these messages can come from untrusted
+parties -- including attackers (e.g., the so-called "coffee shop" attack) that
+leverage many users' lack of a nuanced model of the trust relationships between
+all of the parties that are involved in the service they are using. Furthermore,
+lowering the barrier to the presentation of messages explaining why access has
+been denied by the DNS resolver risks encouraging the wider deployment of
+DNS-based censorship on the Internet.
```

Comment:
It might be good to clarify that this paragraph is about what the I-D is _not_ proposing.